src/map/script.c

```diff
@@ -13573,14 +13573,28 @@ BUILDIN(atcommand) {
 
 /*==========================================
  * Displays a message for the player only (like system messages like "you got an apple" )
+ * usage : dispbottom "<message>"{,<color>{,<account ID>}};
  *------------------------------------------*/
-BUILDIN(dispbottom)
-{
-	TBL_PC *sd=script->rid2sd(st);
+BUILDIN(dispbottom) {
+	TBL_PC *sd = NULL;
+	int color = 0;
 	const char *message;
-	message=script_getstr(st,2);
-	if(sd)
-		clif_disp_onlyself(sd,message,(int)strlen(message));
+
+	message = script_getstr(st,2);
+	if (script_hasdata(st,3))
+		color = script_getnum(st,3);
```

Comment:
Does this even work? I can see an overflow if you use a color with red value greater than `0x7f`.